presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftMetastoreUtil.java

```diff
@@ -399,11 +411,18 @@ public static Table fromMetastoreApiTable(org.apache.hadoop.hive.metastore.api.T
             throw new PrestoException(HIVE_INVALID_METADATA, "Table is missing storage descriptor");
         }
 
+        // Materialized view fetched from Metastore in Facebook uses MANAGED_TABLE table type. Cast it back to MATERIALIZED_VIEW here.
```

Comment:
Don't mention Facebook. This is open source; we should be neutral. Add a TODO here to remove the type change after Hive 3.0 upgrade